core/translation.h

```diff
@@ -39,27 +42,48 @@ class Translation : public Resource {
 	RES_BASE_EXTENSION("translation");
 
 	String locale = "en";
-	Map<StringName, StringName> translation_map;
+	int plural_forms = 0; // 0 means no "Plural-Forms" is given in the PO header file. The min for all languages is 1.
+	String plural_rule;
+
+	// TLDR: Maps context to a list of source strings and translated strings. In PO terms, maps msgctxt to a list of msgid and msgstr.
+	// The first key corresponds to context, and the second key (of the contained HashMap) corresponds to source string.
+	// The value Vector<StringName> in the second map stores the translated strings. Index 0, 1, 2 matches msgstr[0], msgstr[1], msgstr[2]... in the case of plurals.
+	// Otherwise index 0 mathes to msgstr in a singular translation.
+	// Strings without context have "" as first key.
+	HashMap<StringName, HashMap<StringName, Vector<StringName>>> translation_map;
```

Comment:
It would be nice to try to measure the performance impact of using a more complex data structure on a somewhat msg-heavy test project, both for PO files and CSV files.

That would help validate that this is the right approach, compared to e.g. using separate `Translation` resources for each context for example.